### PR TITLE
Refactor plan bootstrap

### PR DESCRIPTION
**Pull request by nickpell**
A prefactor for changing the `SoftwarePlanVersion` model (https://github.com/dimagi/commcare-hq/pull/9781)

### CODE REVIEW

File: corehq/apps/accounting/management/commands/cchq_software_plan_bootstrap.py [hunk 1]
```diff
@@ -74,248 +79,257 @@ def handle(self, dry_run=False, verbose=False, fresh_start=False, flush=False, f
...
+            role_slug = BOOTSTRAP_EDITION_TO_ROLE[edition]
+            try:
+                role = Role.objects.get(slug=role_slug)
+            except ObjectDoesNotExist:
+                logger.info("Could not find the role '%s'. Did you forget to run cchq_prbac_bootstrap?")
+                logger.info("Aborting. You should figure this out.")
+                return
+            software_plan_version = SoftwarePlanVersion(role=role)
+
+            product, product_rates = _ensure_product_and_rate(product_type, edition, dry_run=dry_run, verbose=verbose, apps=apps)
```
Comment (dimagimon) [inline corehq/apps/accounting/management/commands/cchq_software_plan_bootstrap.py]: E501 line too long (129 > 115 characters)

File: corehq/apps/accounting/management/commands/cchq_software_plan_bootstrap.py [hunk 2]
```diff
@@ -74,248 +79,257 @@ def handle(self, dry_run=False, verbose=False, fresh_start=False, flush=False, f
...
+            feature_rates = _ensure_feature_rates(edition_to_features[edition], edition, dry_run=dry_run, verbose=verbose, for_tests=for_tests, apps=apps)
```
Comment (dimagimon) [inline corehq/apps/accounting/management/commands/cchq_software_plan_bootstrap.py]: E501 line too long (154 > 115 characters)

File: corehq/apps/accounting/management/commands/cchq_software_plan_bootstrap.py [hunk 3]
```diff
@@ -74,248 +79,257 @@ def handle(self, dry_run=False, verbose=False, fresh_start=False, flush=False, f
...
+                        feature_rate.save()
+                        software_plan_version.feature_rates.add(feature_rate)
+                    software_plan_version.save()
+
+            if edition == SoftwarePlanEdition.ADVANCED:
+                trials = [True, False]
+            else:
+                trials = [False]
+            for is_trial in trials:
+                default_product_plan = DefaultProductPlan(product_type=product.product_type, edition=edition, is_trial=is_trial)
```
Comment (dimagimon) [inline corehq/apps/accounting/management/commands/cchq_software_plan_bootstrap.py]: E501 line too long (128 > 115 characters)

**Comment by nickpell** [comment]
@biyeun @benrudolph @proteusvacuum cc: @emord

**Comment by benrudolph** [comment]
nice good cleanup. easier with `w=1`. :haircut:

**Pull request merged**